### PR TITLE
[Bug](Decimalv3) coredump of decimalv3 multiply

### PR DESCRIPTION
**Pull request by HappenLee**
# Proposed changes

Issue Number: close #xxx

## Problem summary

Describe your changes.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [ ] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [ ] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [ ] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 36.55 seconds
 load time: 659 seconds
 storage size: 17123117276 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20221228191222_clickbench_pr_70481.html

File: be/src/vec/data_types/data_type_decimal.h [hunk 1]
```diff
@@ -219,56 +219,29 @@ class DataTypeDecimal final : public IDataType {
...
-        const DataTypeDecimal<T>& tx, const DataTypeDecimal<U>& ty, bool is_multiply,
-        bool is_divide) {
-    if constexpr (IsDecimalV2<T> && IsDecimalV2<U>) {
-        return DataTypeDecimal<U>(max_decimal_precision<U>(), 9);
-    } else {
-        UInt32 scale = (tx.get_scale() > ty.get_scale() ? tx.get_scale() : ty.get_scale());
+        size_t multiply_divide_precision = tx.get_precision() + ty.get_precision();
+        size_t plus_minus_precision =
+                std::max(tx.get_precision() - tx.get_scale(), ty.get_precision() - ty.get_scale()) +
+                scale;
```
Comment (Gabriel39) [inline be/src/vec/data_types/data_type_decimal.h:234]: should be max(tx.get_scale(), ty.get_scale())

File: be/src/vec/data_types/data_type_decimal.h [hunk 2]
```diff
@@ -219,56 +219,29 @@ class DataTypeDecimal final : public IDataType {
...
         if (is_multiply) {
             scale = tx.get_scale() + ty.get_scale();
+            precision = std::min(multiply_divide_precision, max_decimal_precision<Decimal128I>());
         } else if (is_divide) {
             scale = tx.get_scale();
+            precision = std::min(multiply_divide_precision, max_decimal_precision<Decimal128I>());
```
Comment (Gabriel39) [inline be/src/vec/data_types/data_type_decimal.h:240]: precision for division should be tx.get_precision() + ty.get_scale()

File: be/src/vec/data_types/data_type_decimal.h [hunk 3]
```diff
@@ -219,56 +219,29 @@ class DataTypeDecimal final : public IDataType {
...
+        } else if (is_plus_minus) {
+            precision = std::min(plus_minus_precision, max_decimal_precision<Decimal128I>());
```
Comment (Gabriel39) [inline be/src/vec/data_types/data_type_decimal.h:242]: forget to set scale?

File: be/src/vec/data_types/data_type_decimal.h (hunk 1 above)
Comment (HappenLee) [reply be/src/vec/data_types/data_type_decimal.h:234]: scale = max(tx.get_scale(), ty.get_scale())

**Review by Gabriel39 — APPROVED** [review]
LGTM

File: be/src/vec/data_types/data_type_decimal.h (hunk 3 above)
Comment (HappenLee) [reply be/src/vec/data_types/data_type_decimal.h:243]: use default max

**Pull request merged**